src/test/test_step.go

```diff
@@ -310,6 +313,7 @@ func testCommandAndEnv(state *core.BuildState, target *core.BuildTarget, run int
 	if len(state.TestArgs) > 0 {
 		args := strings.Join(state.TestArgs, " ")
 		replacedCmd += " " + args
+		args = strings.Replace(args, " ", "|", -1)
```

Comment:
I don't think it's valid to just do this as a global replacement on whatever the user passed in.

Can we move this somewhere that's more specific to `--failed`?

Reply:
It's a problem regardless of the `--failed` flag when you provide more than one test argument to test, e.g. `plz test TARGET ARG1 ARG2`, which is what `--failed` creates internally.

Reply:
but what if I *want* to pass "ARG1 ARG2" to my test? Not all tests are Go tests and they are allowed to accept other flags, I don't think plz can have a hard policy here that it always makes this replacement.

Reply:
Ah right I see what you mean!
Is the `TESTS` env used in all tests or just Go? I've only seen it used in `please_go_test`. In any case I guess I can just do this replacement in the `testMainImpl` of Go tests only, since that is the problem I've observed.

Reply:
the args are definitely consumed by Python tests too, but I can't remember if they do it through that env var or through argv

Reply:
Can't find a reference in the code. It seems `unittest` requires the format of args to be space separated but it might be through the command directly. There is also a `TEST_ARGS` env that is set when creating the environment but I can't see it being used either.